session.go

```diff
@@ -448,6 +449,10 @@ func (s *session) retry(goCtx goctx.Context, maxCnt int, infoSchemaChanged bool)
 				break
 			}
 		}
+		if ctx.Value("mockRetryForTest") != nil {
+			err = kv.ErrRetryable
+			ctx = goctx.Background()
```

Comment:
Why create a new ctx?

Reply:
If we don't reset the ctx, the mock code will run to here all the time and dead loop...
Here it means set a mock flag, do something, clear the flag as we've done what we need.